app/models/package_manager/nu_get.rb

```diff
@@ -226,5 +232,19 @@ def self.parse_requirements(range)
       requirements << ">= 0" if requirements.empty?
       requirements.join(" ")
     end
+
+    # Returns canonical casing for case-insensitive NuGet package names
+    def self.fetch_canonical_nuget_name(name)
+      nuget_href = "https://www.nuget.org/packages/#{name}"
+      page = get_html(nuget_href)
+      og_url_element = page.css("meta[property='og:url']").first
+
+      unless og_url_element
+        StructuredLog.capture("FETCH_CANONICAL_NAME_FAILED", { platform: "nuget", name: name })
+        return false
+      end
+
+      og_url_element.attributes.fetch("content").text.sub("https://nuget.org/packages/", "").sub(/\/$/, "")
```

Comment:
What happens if the upstream changes where they put this field / html structure?

Reply:
I want it to fail loudly or sound an alert when there's a spike in fetch errors. I can make the failure more explicit here though

Reply:
HTML parsing is pretty fraught with errors. The only bad case I think we can autodetect is if we somehow manage to extract some sort of empty string / nil without erroring, which I think we don't ever want to map to anything, so failing loudly seems like a good idea.